rpc/flipt/marshaller.go

```diff
@@ -34,8 +36,27 @@ func (m *V1toV2MarshallerAdapter) Marshal(v interface{}) ([]byte, error) {
 	return m.JSONPb.Marshal(v)
 }
 
+// decoderInterceptor intercepts and modifies the outbound error return value for
+// inputs that fail to unmarshal against the protobuf.
+type decoderInterceptor struct {
+	grpc_gateway_v1.Decoder
+}
+
+func (c *decoderInterceptor) Decode(v interface{}) error {
+	err := c.Decoder.Decode(v)
+	if err != nil {
+		if _, ok := err.(*json.UnmarshalTypeError); ok {
+			return fmt.Errorf("invalid values for key(s) in json body")
```

Comment:
This looks great. One extra suggestion: could we slip in a logger and log line in here?
Something to emit the original error out via logs.

`error` level is the obvious choice. However, I wonder if this actually makes more sense at a `debug` level.
Just thinking if someone spams the API with a bad request then `debug` level might be more forgiving.

Reply:
@GeorgeMac Agreed, especially if someones client is configured badly. I can emit a debug level log here first than we can adjust if necessary.

Reply:
💯 sounds great!